tests/testsuite/death.rs

```diff
@@ -88,32 +88,111 @@ fn ctrl_c_kills_everyone() {
 }
 
 #[cargo_test]
-fn kill_never_corrupts_cargo_toml() {
-    use std::str;
+fn kill_cargo_add_never_corrupts_cargo_toml() {
+    cargo_test_support::registry::init();
+    cargo_test_support::registry::Package::new("my-package", "0.1.1+my-package").publish();
 
     let with_dependency = r#"
-    [package]
-    name = "foo"
-    version = "0.0.1"
-    authors = []
-    build = "build.rs"
-
-    [dependencies]
-    bar = "0.0.1"
-    "#;
+[package]
+name = "foo"
+version = "0.0.1"
+authors = []
+
+[dependencies]
+my-package = "0.1.1"
+"#;
+    let without_dependency = r#"
+[package]
+name = "foo"
+version = "0.0.1"
+authors = []
+"#;
+
+   for i in [30, 60, 90] {
+        let p = project()
+            .file("Cargo.toml", without_dependency)
+            .file("src/lib.rs", "")
+            .build();
+
+        let mut cargo = p.cargo("add").arg("my-package").build_command();
+        cargo
+            .stdin(Stdio::piped())
+            .stdout(Stdio::piped())
+            .stderr(Stdio::piped());
+
+        // spawn cargo
+        let mut child = cargo.spawn().unwrap();
+
+        // nap for a slightly random amount of time.
+        thread::sleep(time::Duration::from_millis(i));
+
+        // kill the process unexpectedly
+        let kill_result = child.kill();
+        assert!(kill_result.is_ok());
+
+        // collect the process
+        let wait_result = child.wait();
+        assert!(wait_result.is_ok());
```

Comment:
nit: these comments seep to be repeating exactly what the code says

Reply:
I'm still new to rust and it helps me sometimes to be able to squint my eyes and just see the comments.  Agree it's redundant here. Tidy.